src/Paystack.php

```diff
@@ -127,9 +127,23 @@ public function setResponse($relativeUrl)
         return $this;
     }
 
-    private function setGetResponse($relativeUrl)
+    private function setGetResponse($relativeUrl, $body = [])
     {
-        $this->response = $this->client->get($this->baseUrl . $relativeUrl, []);
+        $this->response = $this->client->get($this->baseUrl . $relativeUrl, $body);
+
+        return $this;
+    }
+
+    private function setPostResponse($relativeUrl, $body = [])
+    {
+        $this->response = $this->client->post($this->baseUrl . $relativeUrl, $body);
+
+        return $this;
+    }
+
+    private function setPutResponse($relativeUrl, $body = [])
+    {
+        $this->response = $this->client->put($this->baseUrl . $relativeUrl, $body);
```

Comment:
Now all these 3 can be just one method. Right now, the three methods just makes it redundant. It can be like so:

```php
 private function setHttpResponse($relativeUrl, $method = null, $body = [])
{ 
  if (is_null($method) {
     // throw the isNull exception
  }

  $this->response = $this->client->{strtolower($method)}($this->baseUrl . $relativeUrl, $body);

  return $this;
}